docs/fsharp/style-guide/formatting.md

```diff
@@ -1040,210 +852,408 @@ with
-
-let function1 arg1 arg2 arg3 arg4 =
-    arg1 + arg2 +
-    arg3 + arg4
+[<NoEquality; NoComparison>]
+type SynBinding =
+    | SynBinding of
+        accessibility: SynAccess option *
+        kind: SynBindingKind *
+        mustInline: bool *
+        isMutable: bool *
+        attributes: SynAttributes *
+        xmlDoc: PreXmlDoc *
+        valData: SynValData *
+        headPat: SynPat *
+        returnInfo: SynBindingReturnInfo option *
+        expr: SynExpr *
+        range: range *
+        seqPoint: DebugPointAtBinding
 ```
 
-### Formatting pipeline operators or mutable assignments
-
-Pipeline `|>` operators should go underneath the expressions they operate on.
+When documentation comments are added, use an empty line before each `///` comment.
 
 ```fsharp
-// Preferred approach
-let methods2 =
-    System.AppDomain.CurrentDomain.GetAssemblies()
-    |> List.ofArray
-    |> List.map (fun assm -> assm.GetTypes())
-    |> Array.concat
-    |> List.ofArray
-    |> List.map (fun t -> t.GetMethods())
-    |> Array.concat
+/// The volume
+type Volume =
 
-// Not OK
-let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
-            |> List.ofArray
-            |> List.map (fun assm -> assm.GetTypes())
-            |> Array.concat
-            |> List.ofArray
-            |> List.map (fun t -> t.GetMethods())
-            |> Array.concat
+    /// The volume in liters
+    | Liter of float
 
-// Not OK either
-let methods2 = System.AppDomain.CurrentDomain.GetAssemblies()
-               |> List.ofArray
-               |> List.map (fun assm -> assm.GetTypes())
-               |> Array.concat
-               |> List.ofArray
-               |> List.map (fun t -> t.GetMethods())
-               |> Array.concat
+    /// The volume in fluid ounces
+    | FluidOunce of float
+
+    /// The volume in imperial pints
+    | ImperialPint of float
 ```
 
-This also applies to mutable setters:
+### Formatting literal declarations
+
+[F# literals](../language-reference/literals.md) using the `Literal` attribute should place the attribute on its own line and use PascalCase naming:
 
 ```fsharp
-// Preferred approach
-ctx.Response.Headers.[HeaderNames.ContentType] <-
-    Constants.jsonApiMediaType |> StringValues
-ctx.Response.Headers.[HeaderNames.ContentLength] <-
-    bytes.Length |> string |> StringValues
+[<Literal>]
+let Path = __SOURCE_DIRECTORY__ + "/" + __SOURCE_FILE__
 
-// Not OK
-ctx.Response.Headers.[HeaderNames.ContentType] <- Constants.jsonApiMediaType
-                                                  |> StringValues
-ctx.Response.Headers.[HeaderNames.ContentLength] <- bytes.Length
-                                                    |> string
-                                                    |> StringValues
+[<Literal>]
+let MyUrl = "www.mywebsitethatiamworkingwith.com"
 ```
 
-### Formatting modules
+Avoid placing the attribute on the same line as the value.
+
+## Formatting module declarations
```

Comment:
Should this be an H3 since it falls under broader "Formatting declarations" section?